src/core/load-dropin.c

```diff
@@ -19,51 +19,245 @@
 
 
 #include "conf-parser.h"
+#include "fs-util.h"
```

Comment:
I just realized that the first weird case:

> a .wants symlink to /dev/null was masking the unit the symlink was refering to

is not correct.

It masks a unit only if this unit is not defined at all. Only `systemctl status` will report the unit masked but it doesn't have any other bad side effects.